vscode-test-runner/workspace-1/.joyride/src/integration_test/index.cljs

```diff
@@ -0,0 +1,52 @@
+(ns integration-test.index
+  (:require [cljs.test]
+            [promesa.core :as p]))
+
+(def !results (atom {:fail 0
+                     :error 0}))
+
+(defmethod cljs.test/report [:cljs.test/default :begin-test-var] [m]
+  (println "===" (-> m :var meta :name))
+  (println))
+
+(def old-fail (get-method cljs.test/report [:cljs.test/default :fail]))
+
+(defmethod cljs.test/report [:cljs.test/default :fail] [m]
+  (old-fail m)
+  (swap! !results update :fail inc))
+
+(def old-error (get-method cljs.test/report [:cljs.test/default :fail]))
+
+(defmethod cljs.test/report [:cljs.test/default :error] [m]
+  (old-error m)
+  (swap! !results update :error inc))
+
+(defn run-all-tests []
+  (p/create (fn [resolve, reject]
+              (p/do!
+               (p/delay 1000) ; This waits for workspace_activate.cljs to run
+               (require '[integration-test.workspace-activate-test])
+
+               ;; This fails when accessing integration-test.run-a-ws-script-test/symbol-1
+               ;; if required from the ns form
+               (require '[integration-test.ws-scripts-test])
+               #_(require '[integration-test.npm-test])
+
+               (cljs.test/run-tests 'integration-test.workspace-activate-test)
```

Comment:
I don't think running tests like this is going to work properly. `cljs.test/run-tests` runs tests asynchronously but the API doesn't return a promise.
Instead, I think it's better to only "fire" `cljs.test/run-tests` once. You can append the symbols of those namespaces to the one call together.

Reply:
✅